src/map/pc.cpp

```diff
@@ -9534,6 +9534,13 @@ TIMER_FUNC(pc_close_npc_timer){
 void pc_close_npc(map_session_data *sd,int flag)
 {
 	nullpo_retv(sd);
+	
+	for (const auto &it : sd->npc_id_dynamic) {
```

Comment:
This only closes a single NPC, why do we update all?

Reply:
Actually I'm not sure about this part. I wanted to initialize dynamic timer when NPC secure time expires. Please suggest if there is a better way.

Reply:
I would assume if you're using the NPC Secure Timer then that should only apply to the NPC the player is talking to from the `npc_id_dynamic` list.

Reply:
@aleos89 Thanks!